batch/batch/cloud/azure/driver/pricing.py

```diff
@@ -0,0 +1,165 @@
+import abc
+from typing import List, Optional, Dict
+
+import datetime
+import dateutil.parser
+import asyncio
+import logging
+
+from hailtop.aiocloud import aioazure
+from hailtop.utils import flatten, grouped, time_msecs, parse_timestamp_msecs
+
+from ..resource_utils import azure_valid_machine_types, azure_disk_name_to_storage_gib
+
+
+log = logging.getLogger('pricing')
+
+
+class AzurePrice(abc.ABC):
+    region: str
+    effective_start_date: int
+    effective_end_date: Optional[int]
+    time_updated: int
+
+    def is_current_price(self):
+        now = time_msecs()
+        return (now >= self.effective_start_date
+                and (self.effective_end_date is None or now <= self.effective_end_date))
+
+    @property
+    def version(self) -> str:
+        return datetime.datetime.utcfromtimestamp(self.effective_start_date / 1000).strftime('%Y-%m-%d')
```

Comment:
It makes me a bit uncomfortable that we throw away information here. Why don't we just use the actual effective_start_date in (it seems) milliseconds since the epoch as the version? I feel like there's all sorts of opportunities for confusion if Azure ever sends us a start date that's, say, at noon. Is there a guarantee in their documentation that the start date is always midnight or something?

Reply:
We can do the full timestamp. We might want a resource_versions table instead that contains when the version was added. That would fulfill the purpose of easily being able to tell what's going on.

However, you said something about not having a boolean field to say which is the latest version.